File: app/src/main/java/quintype/com/templatecollectionwithrx/models/story/UserEngagement.java
package quintype.com.templatecollectionwithrx.models.story;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

import java.util.List;

/**
 * An object representing UserEngagement for story.
 *
 * @author Imran imran@quintype.com
 * @author Madhu madhu@quintype.com
 */
public class UserEngagement implements Parcelable {

    @SerializedName("story-content-id")
    private String storyContentId;
    @SerializedName("votes")
    private List<String> votes;
    @SerializedName("comments")
    private List<Comment> comments;

    public String storyContentId() {
        return storyContentId;
    }

    public List<String> votes() {
        return votes;
    }

    public List<Comment> comments() {
        return comments;
    }

    @Override
    public String toString() {
        return "UserEngagement{" +
                "storyContentId='" + storyContentId + '\'' +
                ", votes=" + votes +
                ", comments=" + comments +
                '}';
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.storyContentId);
        dest.writeStringList(this.votes);
        dest.writeTypedList(comments);
    }

    public UserEngagement() {
    }

    protected UserEngagement(Parcel in) {
        this.storyContentId = in.readString();
        this.votes = in.createStringArrayList();
        this.comments = in.createTypedArrayList(Comment.CREATOR);
    }

    public static final Creator<UserEngagement> CREATOR = new Creator<UserEngagement>() {
        public UserEngagement createFromParcel(Parcel source) {
            return new UserEngagement(source);
        }

        public UserEngagement[] newArray(int size) {
            return new UserEngagement[size];
        }
    };
}

File: app/src/main/java/quintype/com/templatecollectionwithrx/models/story/UserEngagement.kt
package quintype.com.templatecollectionwithrx.models.story

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.SerializedName

/**
 * Created TemplateCollectionWithRx by rakshith on 9/4/18.
 */

class UserEngagement : Parcelable {

    @SerializedName("story-content-id")
    private val storyContentId: String
    @SerializedName("votes")
    private val votes: List<String>
    @SerializedName("comments")
    private val comments: List<Comment>

    override fun toString(): String {
        return "UserEngagement{" +
                "storyContentId='" + storyContentId + '\''.toString() +
                ", votes=" + votes +
                ", comments=" + comments +
                '}'.toString()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(this.storyContentId)
        dest.writeStringList(this.votes)
        dest.writeTypedList(comments)
    }

    protected constructor(`in`: Parcel) {
        this.storyContentId = `in`.readString()
        this.votes = `in`.createStringArrayList()
        this.comments = `in`.createTypedArrayList(Comment)
    }

    companion object CREATOR : Parcelable.Creator<UserEngagement> {
        override fun createFromParcel(source: Parcel): UserEngagement {
            return UserEngagement(source)
        }

        override fun newArray(size: Int): Array<UserEngagement?> {
            return arrayOfNulls(size)
        }
    }
}
